sceptre/cli/__init__.py

```diff
@@ -63,7 +63,17 @@ def cli(
     }
     user_variables = {}
     if var_file:
-        user_variables.update(yaml.safe_load(var_file.read()))
+        for fh in var_file:
+            parsed = yaml.safe_load(fh.read())
+            # intersection
+            overloaded_keys = set(user_variables.keys()) & set(parsed.keys())
+            if overloaded_keys:
+                logger.debug(
+                    "Duplicate variables encountered: {0}. "
+                    "Using values from file with highest precedence: '{1}'."
```

Comment:
Not sure we need `Using values from file with highest precedence: ` as it makes the message overly verbose and is pretty easy to which file is taking precedence.

Reply:
Is the verbosity that much of an issue? I suspect it may be more helpful when there were unexpectedly repeated variables (the reasoning is similar to why I advocated for erroring out for duplicates) and it informs the user how we're selecting precedence without having to point them to a guide, which is a decently big usability gain. Especially since this is debug logging which is already verbose, anyway, adding all relevant information seems sensible to me.

Reply:
Could we convey the same information with a simpler sentence?

`"Duplicate variables encountered: {}. Using values from {}".format(', '.join(overloaded_keys), fh.name)`

Reply:
After this change I'm happy with it!

Reply:
done!